jax/_src/tree_util.py

```diff
@@ -46,56 +50,59 @@ def tree_flatten(tree, is_leaf: Optional[Callable[[Any], bool]] = None):
       stopped immediately, with the whole subtree being treated as a leaf.
 
   Returns:
-    A pair where the first element is a list of leaf values and the second
+    A pair where the first element is a sequence of leaf values and the second
     element is a treedef representing the structure of the flattened tree.
   """
   return pytree.flatten(tree, is_leaf)
 
 
-def tree_unflatten(treedef, leaves):
+def tree_unflatten(treedef: PyTreeDef, leaves: Iterable[Array]) -> Any:
   """Reconstructs a pytree from the treedef and the leaves.
 
   The inverse of :func:`tree_flatten`.
 
   Args:
     treedef: the treedef to reconstruct
-    leaves: the list of leaves to use for reconstruction. The list must match
-      the leaves of the treedef.
+    leaves: the iterable of leaves to use for reconstruction. The iterable
+      must match the leaves of the treedef.
 
   Returns:
     The reconstructed pytree, containing the ``leaves`` placed in the structure
     described by ``treedef``.
   """
   return treedef.unflatten(leaves)
 
-def tree_leaves(tree, is_leaf: Optional[Callable[[Any], bool]] = None):
+def tree_leaves(tree: Any,
+                is_leaf: Optional[Callable[[Any], bool]] = None) -> (
+                    Sequence[Array]):
```

Comment:
```suggestion
                is_leaf: Optional[Callable[[Any], bool]] = None
                ) -> Sequence[Array]:
```

Reply:
As mentioned above, I think we should annotate the return as `List[Array]` instead of `Sequence[Array]`.